src/mcp/server/auth/routes.py

```diff
@@ -147,31 +147,19 @@ def create_auth_routes(
     return routes
 
 
-def modify_url_path(url: AnyHttpUrl, path_mapper: Callable[[str], str]) -> AnyHttpUrl:
-    return AnyHttpUrl.build(
-        scheme=url.scheme,
-        username=url.username,
-        password=url.password,
-        host=url.host,
-        port=url.port,
-        path=path_mapper(url.path or ""),
-        query=url.query,
-        fragment=url.fragment,
-    )
-
-
 def build_metadata(
     issuer_url: AnyHttpUrl,
     service_documentation_url: AnyHttpUrl | None,
     client_registration_options: ClientRegistrationOptions,
     revocation_options: RevocationOptions,
 ) -> OAuthMetadata:
-    authorization_url = modify_url_path(
-        issuer_url, lambda path: path.rstrip("/") + AUTHORIZATION_PATH.lstrip("/")
+    authorization_url = AnyHttpUrl(
+        str(issuer_url).rstrip("/") + AUTHORIZATION_PATH
```

Comment:
There's no need to strip in any of those.

```suggestion
        urljoin(str(issuer_url), AUTHORIZATION_PATH)
```

Reply:
Thanks for the suggestion, @Kludex!

I initially thought the same — that `urljoin(str(issuer_url), "/authorize")` would be enough. But it actually breaks in some cases where the `base_url` includes a path.

For example, when the `base_url` is `https://example.com/auth/oidc/op/Customer/`, `urljoin` returns `https://example.com/authorize`, which drops the intended path entirely. That’s because `urljoin` treats the `/authorize` as an absolute path and replaces everything after the domain.

To illustrate this, I added a test suite comparing both approaches:

```python
import unittest
from urllib.parse import urljoin
from pydantic import AnyHttpUrl

class TestModifyUrlPath(unittest.TestCase):
    def test_append_authorize_to_urls_with_urljoin(self):
        """Test appending /authorize to various URL formats using urljoin"""
        test_cases = [
            ("https://example.com", "https://example.com/authorize"),
            ("https://example.com/", "https://example.com/authorize"),
            ("https://example.com/auth/oidc/op/Customer", "https://example.com/auth/oidc/op/Customer/authorize"),
            ("https://example.com/auth/oidc/op/Customer/", "https://example.com/auth/oidc/op/Customer/authorize"),
            ("http://localhost:8000", "http://localhost:8000/authorize"),
            ("http://localhost:8000/", "http://localhost:8000/authorize"),
        ]

        for base_url, expected in test_cases:
            any_http_url = AnyHttpUrl(base_url)
            with self.subTest(base_url=any_http_url):
                result = AnyHttpUrl(urljoin(str(any_http_url), "/authorize"))
                self.assertEqual(result, AnyHttpUrl(expected))

    def test_append_authorize_to_urls_with_rstrip(self):
        """Test appending /authorize to various URL formats using rstrip"""
        test_cases = [
            ("https://example.com", "https://example.com/authorize"),
            ("https://example.com/", "https://example.com/authorize"),
            ("https://example.com/auth/oidc/op/Customer", "https://example.com/auth/oidc/op/Customer/authorize"),
            ("https://example.com/auth/oidc/op/Customer/", "https://example.com/auth/oidc/op/Customer/authorize"),
            ("http://localhost:8000", "http://localhost:8000/authorize"),
            ("http://localhost:8000/", "http://localhost:8000/authorize"),
        ]

        for base_url, expected in test_cases:
            any_http_url = AnyHttpUrl(base_url)
            with self.subTest(base_url=any_http_url):
                result = AnyHttpUrl(str(any_http_url).rstrip("/") + "/authorize")
                self.assertEqual(result, AnyHttpUrl(expected))

if __name__ == "__main__":
    unittest.main()
```

So for consistency across all cases, I believe we need to keep the rstrip("/") approach.

Let me know what you think!